knover/data/dialog_reader.py

```diff
@@ -338,7 +338,7 @@ def _read_numerical_file(self, fp, phase, is_infer, delimiter=";"):
                 # if you change the numerical data format, you must to make sure the last part of
                 # numerical sequence is the target sequence
                 def rindex(lst, elem):
-                    return len(lst) - lst[-1:0:-1].index(elem) - 1
+                    return len(lst) - lst[::-1].index(elem) - 1
```

Comment:
Get the last position of [BOS] token, which is the start position of text generation.
It is also suitable for autoregressive task.